### ISSUE TITLE
Support transferrable objects with Bun.spawn IPC ("Unable to deserialize data" when sending `MessagePort`)

### ISSUE THREAD

**Issue by futpib** [oven-sh/bun#8156]
Docs claim that you can send transferable types to child processes:

https://github.com/oven-sh/bun/blob/86b1aa02b3be3076cd9d18dbc301e648500ad6c1/docs/api/spawn.md?plain=1#L230

With a link to MDN https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Transferable_objects#supported_objects

Which says `MessagePort` is transferrable.

In fact when you try this you get `TypeError: Unable to deserialize data.`

**Comment by paperclover** [comment]
serialization across subprocesses has never supported transferrable objects. we shouldn't be linking to that part of the docs.

**Comment by paperclover** [comment]
i dont think this is trivial in many many cases, because memory related to a transferrable object would have to be shared across processes; and in the case of `MessagePort` it would require a separate implementation that uses the IPC for messages.
